util/types/time.go

```diff
@@ -471,6 +471,14 @@ func (t *Time) Sub(t1 *Time) Duration {
 	}
 }
 
+// Add adds d to t, returns a duration value.
+// Note that add should not be done on different time types.
+func (t *Time) Add(d *Duration) Duration {
+	d.Duration = gotime.Duration(-int64(d.Duration))
+	t2, _ := d.ConvertToTime(t.Type)
```

Comment:
The implementation of function `(t *Time)Add(d *Duration)` use function `(t *Time)Sub(t2 *Time)` , the work of these check would be done in `Sub`.